### DISCUSSION TITLE
HomeAssistant OS in a Jail

### DISCUSSION THREAD

**Discussion by haldi4803** [Jip-Hop/jailmaker#152] [Q&A]
Okay, Title says everything right?
[HomeAssistant OS](https://github.com/home-assistant/operating-system) runs on Debian 12 and also uses Systemd, therefore it SHOULD work right?
There is a way to install [Home Assistant Supervised](https://github.com/home-assistant/supervised-installer) on a normal Debian Bookworm system. But that looks really complicated and seems to provide a lot of failure points... i'd rather not try that.

So how do we get the HAOS distribution included in Jailmaker then?
I've seen that the "create new jail" function uses the [lxc-download.in](https://github.com/Jip-Hop/lxc/blob/master/templates/lxc-download.in) to look for distros. which uses http://images.linuxcontainers.org/ to search for images.
But then again jailmaker uses systemd-nspawn and not lxc right?
So those images get converted/unpacked into what exactly?
Maybe there is a build of HomeAssistantOS that gets build similar and can be used.

Or am i looking at this from the wrong perspective? is there an easier way to import a custom OS? in What format would that need to be?

Regards,

**Comment by Jip-Hop** [comment]
You need a plain rootfs. You can look around in the jails directory. Each jail has a rootfs dir. For details like this I recommend to study the code of jailmaker.

**Comment by BenRoe** [comment]
I am also interested.

**Comment by Jip-Hop** [comment]
The Supervised install doesn't look that complicated actually: https://github.com/home-assistant/supervised-installer?tab=readme-ov-file

You could try those steps in a debian bookworm jail. Ensure you're using macvlan or bridge networking. You may need to disable seccomp filtering and add capabilities.

But I don't think it's a good idea to run HAOS in a jail and I doubt it will work. HOAS is not designed to run in a jail (lxc or otherwise) as far as I know. So you're asking for trouble if you take this approach...

https://community.home-assistant.io/t/running-hassos-as-an-lxd-lxc-virtual-machine/227643/31

**Reply by haldi4803** [reply]
Yeah... coding isn't really my forte. I can barely read a little, but thats it.

So I tried the supervised installer. 
But it fails at the Debian package install.

```
2024-04-29 22:19:00 (54.7 MB/s) - ‘homeassistant-supervised.deb’ saved [7428/7428]

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'homeassistant-supervised' instead of './homeassistant-supervised.deb'
The following NEW packages will be installed:
  homeassistant-supervised
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/7,428 B of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 /home/homeassistant-supervised.deb homeassistant-supervised all 1.7.0 [7,428 B]
debconf: delaying package configuration, since apt-utils is not installed
(Reading database ... 19792 files and directories currently installed.)
Preparing to unpack .../homeassistant-supervised.deb ...
[warn]
[warn] If you want more control over your own system, run
[warn] Home Assistant as a VM or run Home Assistant Core
[warn] via a Docker container.
[warn]
[warn] ModemManager service is enabled. This might cause issue when using serial devices.
[info] Fix kernel dmesg restriction
/var/lib/dpkg/tmp.ci/preinst: line 50: /proc/sys/kernel/dmesg_restrict: Read-only file system
dpkg: error processing archive /home/homeassistant-supervised.deb (--unpack):
 new homeassistant-supervised package pre-installation script subprocess returned error exit status 1
[info] Undo divert on abort-install
No diversion 'diversion of /etc/NetworkManager/NetworkManager.conf to /etc/NetworkManager/NetworkManager.conf.real by homeassistant-supervised', none removed.
No diversion 'diversion of /etc/NetworkManager/system-connections/default to /etc/NetworkManager/system-connections/default.real by homeassistant-supervised', none removed.
No diversion 'diversion of /etc/docker/daemon.json to /etc/docker/daemon.json.real by homeassistant-supervised', none removed.
No diversion 'diversion of /etc/network/interfaces to /etc/network/interfaces.real by homeassistant-supervised', none removed.
[info] Resetting debconf selections
[info] Removal complete, due to the complexity of this installation method,
[info] you will need to manually remove the containers created by the supervisor
Errors were encountered while processing:
 /home/homeassistant-supervised.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
root@HAOS:/home#
```

Is that an issue due "read-only file system" of /proc/sys/kernel/dmesg_restrict?
https://community.home-assistant.io/t/installing-home-assistant-supervised-using-debian-12/200253/905
Reccomends on the Host System do:
```
echo "kernel.dmesg_restrict=0" >> /etc/sysctl.conf
sysctl -p
```
would that be an issue for TrueNAS scale?

**Reply by haldi4803** [reply]
Well i can always just turn it back ^^ so i've tried.
Important note: Change to fixed IP before you install HomeAssistant.... if you do it afterwards it changes network adapter and you can't connect to the web interface anymore -.-

After following the Guide it takes quite a bit until Web interface is ready. But i'm restoring my Backup and wonder what will happen :)

**Reply by haldi4803** [reply]
![image](https://github.com/Jip-Hop/jailmaker/assets/17104473/a301965b-9253-450e-9e0c-e14d2d536a48)
Seems like Docker is installed as [cgroupv2](https://www.home-assistant.io/more-info/unsupported/cgroup_version) and not v1 and i can't change that. Is that also a host setting?




And [network manager](https://www.home-assistant.io/more-info/unsupported/network_manager) doesn't work, thats why the backup failed.
![image](https://github.com/Jip-Hop/jailmaker/assets/17104473/358c0d42-c6d9-4421-bed9-176b090747b5)

```
root@HAOS:~# ha net info
docker:
  address: 172.30.32.0/23
  dns: 172.30.32.3
  gateway: 172.30.32.1
  interface: hassio
host_internet: false
interfaces: []
supervisor_internet: true
```
So i did try to add DNS to nmcli but that's apparently not a persistent method... 
"[then ha network reload && ha network info](https://community.home-assistant.io/t/addonmanager-install-blocked-from-execution-no-host-internet-connection/433134/4?u=haldi)"
Then it switched to `host_internet: true` and i could restore the backup.
But still, the networkManager is wrongly configured....
Edit: `nmcli dev set host0 managed yes` and `service NetworkManager restart`
and after a restart from the Jail the Web interface isn't reachable anymore...



https://community.home-assistant.io/t/journal-issues-with-supervised-install-after-updating-to-2022-11/483879/147
```
root@HAOS:/home# sudo systemctl status systemd-journal-remote
○ systemd-journal-remote.service - Journal Remote Sink Service
     Loaded: loaded (/lib/systemd/system/systemd-journal-remote.service; indirect; preset: disabled)
     Active: inactive (dead)
TriggeredBy: ○ systemd-journal-remote.socket
       Docs: man:systemd-journal-remote(8)
             man:journal-remote.conf(5)
```
Seems to be the issue of not properly mounting Journal "Systemd journal gateway is enabled and mapped into supervisor as /run/systemd-journal-gatewayd.sock"

**Comment by Lockszmith-GH** [comment]
Home Assistant OS is only availabe as a VM, the alternative to running it full fledged is Supervied - I've been running it successfully in a debain VM for years.

The supervied installation is documented very well, and that's probably what you want - a way to mimic the Home Assistant OS in a jail = running supervised.
* [What are supported installation methods](https://github.com/home-assistant/architecture/blob/master/adr/0012-define-supported-installation-method.md)
* [Requirements of the supervised installer](https://github.com/home-assistant/supervised-installer)
* [Supervised Installer](https://github.com/home-assistant/architecture/blob/master/adr/0014-home-assistant-supervised.md)

It looks like an LXC solution does exist and is documented for proxmox users [here](https://github.com/whiskerz007/proxmox_hassio_lxc).

But I'll be weary, while the HA community is great, as HA Supervised is NOT officially supported on LXC, you'll have a hard time asking for assistance.

If you CAN run the VM, that's probably the best solution.

If you DO succeed, I'm more than interested to follow your experience though.

**Comment by cavazos-apps** [comment]
Did you get any further with this?
